src/cart/crt_init.c

```diff
@@ -309,14 +309,14 @@ crt_init_opt(crt_group_id_t grpid, uint32_t flags, crt_init_options_t *opt)
 	struct timeval	now;
 	unsigned int	seed;
 	const char	*path;
-	bool		server;
+	bool		server = false;
 	bool		provider_found = false;
 	int		plugin_idx;
 	int		prov;
 	bool		set_sep = false;
 	int		max_num_ctx = 256;
-	uint32_t	ctx_num;
-	bool		share_addr;
+	uint32_t	ctx_num = 0;
+	bool		share_addr = false;
```

Comment:
It might be clearer to move this to L426.

Reply:
added explicit settings to line 426 as well. i prefer to have it also at the top in case some other code path in future starts using those

Reply:
IMO that's worse.  Now they're set once, then set again before use so there's a risk that the first assignment could be modified yet it would be overwritten by the second one without realising.  The advantage to moving it would be to reduce the scope of the variable so it's clear where it is (and isn't used).

Reply:
ok